mods/persistence/modules/world_save/serializers/sql_serializer.dm

```diff
@@ -408,52 +399,53 @@ var/global/list/serialization_time_spent_type
 		if (!T)
 			to_world_log("Attempting to deserialize onto turf [thing.x],[thing.y],[thing.z] failed. Could not locate turf.")
 			return
-		//T.ChangeTurf(thing.thing_type) //You really don't want to run this before the other SS are initialized!!!!!!!!!
-		//Its referencing SS AO, SS Lighting, various observer events, etc...
-		T.changing_turf = TRUE
-		qdel(T)
-		existing = new thing.thing_type(T)
+		T.ChangeTurf(thing.thing_type)
+		existing = locate(thing.x, thing.y, thing.z)
 	else
 		// default creation
 		existing = new thing.thing_type()
 	existing.persistent_id = thing.p_id // Upon deserialization we reapply the persistent_id in the thing table to save space.
 	reverse_map["[thing.p_id]"] = existing
+
+	//Remove vars not in existing.vars we got in thing.thing_vars
+	for(var/datum/persistence/load_cache/thing_var/TV in thing.thing_vars)
+		if(!(TV.key in existing.vars))
```

Comment:
I know you just moved this from the place below, but instead of doing this, try doing
```
try
    var/V = existing.vars[TV.key]
catch
    log_warning(blah)
```
It should be much faster.

Reply:
That's really not faster? Exceptions are the most expensive thing to handle. Its fast because you probably didn't try it with missing variables, but once it starts throwing exceptions its gonna take centuries to load. Don't ever do flow control with exceptions.

Reply:
I found a way better way to handle them. Just check the loaded vars against the saved vars. Since they're checked at compile time they can't refer to non-existent vars, and its just a regular, fast list lookup.